portal/app/api/v1/endpoint/farm_data_parser.py

```diff
@@ -67,9 +68,13 @@ def main_power_parser(data):
         Parses json blob for device_id = 17 and returns processed data
         Converts power from W to kW and adds field energy to the dataframe
     """
-    farm_data = [ main_farm_parser(row) for row in data ]
-    farm_df = pd.DataFrame(farm_data)
+    farm_data = []
+    for row in data:
+        row_val = main_farm_parser(row)
+        if row_val != None:
```

Comment:
Do some digging as to why you should be using `is not` v.s. `!=`. Let me know what you find.

Reply:
Equality and identity operators and “==” operator compares the values where as “is” operator checks to see if the memory location is the same, therefore the is operation is fundamentally faster. In python there can only be a single none object and therefore all the none values are pointing to the same memory address, which is the reason None should always be compared by the identity operation.

Reply:
I just corrected it.